File: src/modules/object/data/MapObject.js

var SIZE_MAP = 40;

var MapObject = cc.Class.extend(
    {

        name: null,
        type: null,
        id: null,
        position: null,
        size: null,
        state: false,
        imagePath: null,
        beginTime: null,


        ctor: function() {
            this._super()
        },

        onCreate: function() {

        },

        isSelected: function() {

        },

        stopAction: function() {

        },

        executeImmediately: function() {

        },

        sendMessage: function() {

        },

        getName: function() {
            return this.name;
        },

        getType: function() {
            return this.type;
        },

        getId: function() {
            return this.id;
        },

        getPositionX: function() {
            return this.position.x;
        },

        getPositionY: function() {
            return this.position.y;
        },

        getSize: function() {
            return this.size;
        },

        getState: function() {
            return this.state;
        },

        getImage: function() {
            return this.imagePath;
        },

        getBeginTime: function() {
            return this.beginTime;
        },

        setImage: function(imagePath) {
            this.imagePath = imagePath;
        },

        setName: function(name) {
            this.name = name;
        },

        setType: function(type) {
            this.type = type;
        },

        setId: function(id) {
            this.id = id;
        },

        setPosition: function(x, y) {
            this.position.x = x;
            this.position.y = y;
        },

        setSize: function(size) {
            this.size = size;
        },

        setState: function(state) {
            this.state = state;
        },

        setBeginTime: function(beginTime) {
            this.beginTime = beginTime;
        }

    }
)

File: src/modules/object/data/MapObject.java
package modules.object.data;


import modules.game.data.Resource;
import util.database.DataModel;
import java.awt.*;
import java.util.Date;

public class MapObject extends DataModel {

    int typeId;
    String type;
    String typeObj;
    int level;
    String name;
    int size;
    int id;
    Point position;
    boolean state;
    String imagePath;
    Date beginTime;
    Resource cost = new Resource(0, 0, 0);

    public Resource getCost() {
        return cost;
    }

    public void setCost(int gold, int elixir) {
        this.cost.setGold(gold);
        this.cost.setElixir(elixir);
    }

    public int getLevel() {
        return level;
    }

    public void setLevel(int level) {
        this.level = level;
    }


    public String getTypeObj() {
        return typeObj;
    }

    public void setTypeObj(String typeObj) {
        this.typeObj = typeObj;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Point getPosition() {
        return position;
    }

    public void setPosition(Point position) {
        this.position = position;
    }

    public boolean isState() {
        return state;
    }

    public void setState(boolean state) {
        this.state = state;
    }

    public String getImagePath() {
        return imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }

    public Date getBeginTime() {
        return beginTime;
    }

    public void setBeginTime(Date beginTime) {
        this.beginTime = beginTime;
    }

    public int getTypeId() {
        return typeId;
    }

    public void setTypeId(int typeId) {
        this.typeId = typeId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}
